src/main/java/org/humancellatlas/ingest/core/service/MetadataDifferService.java

```diff
@@ -0,0 +1,37 @@
+package org.humancellatlas.ingest.core.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.flipkart.zjsonpatch.JsonDiff;
+import lombok.AllArgsConstructor;
+import org.humancellatlas.ingest.core.JsonPatch;
+import org.humancellatlas.ingest.core.MetadataDocument;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class MetadataDifferService {
```

Comment:
could have a diff() method which returns null if there's no diff and returns patch if there's a diff.

Reply:
The anyDifference() method returns a true/false if there's any diff. generatePatch() returns a JSONPatch(possibly empty, but not null) if there's no diff. I'm not a fan of returning null